src/Service.GraphQLBuilder/GraphQLUtils.cs

```diff
@@ -209,24 +212,38 @@ public static ObjectType UnderlyingGraphQLEntityType(IType type)
         }
 
         /// <summary>
-        /// Parse a given string value to supported GraphQL Type and GraphQLValueNode
+        /// Translates a JSON string or number value defined in the runtime configuration to a GraphQL {Type}ValueNode which represents
+        /// the associated GraphQL type. The target value type is referenced from the passed in parameterDefinition which
+        /// holds database schema metadata.
         /// </summary>
-        public static Tuple<string, IValueNode> GetGraphQLTypeAndNodeTypeFromStringValue(string stringValue)
+        /// <param name="defaultValueFromConfig">String representation of default value defined in runtime config.</param>
+        /// <param name="parameterDefinition">Database schema metadata for stored procedure parameter which include value and value type.</param>
+        /// <returns>Tuple where first item is the string representation of a GraphQLType (e.g. "Byte", "Int", "Decimal")
+        /// and the second item is the GraphQL *type*ValueNode </returns>
+        /// <exception cref="DataApiBuilderException">Raised when parameter casting fails due to unsupported type.</exception>
+        public static Tuple<string, IValueNode> ConvertValueToGraphQLType(string defaultValueFromConfig, ParameterDefinition parameterDefinition)
         {
-            if (int.TryParse(stringValue, out int integerValue))
+            string paramValueType = SchemaConverter.GetGraphQLTypeFromSystemType(type: parameterDefinition.SystemType);
+            Tuple<string, IValueNode> valueNode = paramValueType switch
             {
-                return new(LONG_TYPE, new IntValueNode(integerValue));
-            }
-            else if (double.TryParse(stringValue, out double floatingValue))
-            {
-                return new(FLOAT_TYPE, new FloatValueNode(floatingValue));
-            }
-            else if (bool.TryParse(stringValue, out bool booleanValue))
-            {
-                return new(BOOLEAN_TYPE, new BooleanValueNode(booleanValue));
-            }
+                BYTE_TYPE => new(BYTE_TYPE, new IntValueNode(byte.Parse(defaultValueFromConfig))),
+                SHORT_TYPE => new(SHORT_TYPE, new IntValueNode(short.Parse(defaultValueFromConfig))),
+                INT_TYPE => new(INT_TYPE, new IntValueNode(int.Parse(defaultValueFromConfig))),
+                LONG_TYPE => new(LONG_TYPE, new IntValueNode(long.Parse(defaultValueFromConfig))),
```

Comment:
Is there no LongValueNode?

Reply:
IntValueNode captures all [integral numeric types](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types), added detail in comment: https://github.com/Azure/data-api-builder/pull/1383#discussion_r1153881631